### ISSUE TITLE
Table Sort Button causes AVT 1 DAP Violation - Accessible name must match or contain the visible label text.

### ISSUE THREAD

**Issue by EmerDuffy** [IBM/carbon-components-angular#728]
## Detailed description

The aria label on the table sort button causes a DAP violation as the aria text is not the same or does not contain the column header text e.g. 

**Button Aria Label**
'Sort rows by this header in ascending order'

**Table Header Column Name**
Assessments

To pass it could be... 
'Assessments - Sort rows by this header in ascending order' or
'Assessments - Sort in ascending order' or
'Sort Assessments in ascending order'

This is not only an **AVT 1** issue but also an **AVT 3** issue as the sort buttons won't have unique labels i.e. using the JAWS shortcut to list all buttons every ascending sort button in a table will have the same text therefore there is no way for the user to distinguish between the sort buttons.  The list will appear like this and read out as such:

'Sort rows by this header in ascending order'
'Sort rows by this header in ascending order'
'Sort rows by this header in ascending order'
'Sort rows by this header in ascending order'
'Sort rows by this header in ascending order'

How is a user supposed to know which button is 'Assessments'?

[An accessible name is a short string, typically 1 to 3 words, that authors associate with an element to provide users of assistive technologies with a label for the element.](https://www.w3.org/TR/wai-aria-practices-1.1/#names_and_descriptions_definition)

**Assess Code**

`<button aria-live="polite" class="bx--table-sort bx--table-sort--ascending ng-star-inserted" ng-reflect-klass="bx--table-sort" ng-reflect-ng-class="[object Object]" aria-label="Sort rows by this header in ascending order" data-namewalk="105"><span tabindex="-1" title="Assessments" class="ng-star-inserted" data-namewalk="106"> Assessments </span>...</button>`

**DAP Violation**
Accessible name must match or contain the visible label text.

[Accessible name does not contain the visible text label](https://aat.w3ibm.mybluemix.net/token/a60d9226-20d5-4bb0-aaee-bf788e3b505d/61d96c58-5f80-4c90-bd95-89b28a7d29e2/archives/2019JulyDeploy//doc/w3/help/en-US/idhi_accessibility_check_g1202.html)

> Describe in detail the issue you're having.

The table sort button is causing a DAP violation. 
Accessible name must match or contain the visible label text.

> Is this a feature request (new component, new icon), a bug, or a general issue?
Bug.
Accessibility Issue.

> Is this issue related to a specific component?
Table Sort Button 

> What did you expect to happen? What happened instead? What would you like to see changed?
No AVT 1 DAP violation.
No AVT 3 violation.

> What browser are you working in?
Chrome with DAP Plugin. 

> What version of the Carbon Design System are you using?
    "carbon-components": "10.3.2",
    "carbon-components-angular": "3.14.7",

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
Assess.
Currently trying to pass AVT 1 using DAP plugin. 

## Steps to reproduce the issue
On any table with a sortable column...
Right click - Inspect - Elements Tab - DAP Sub Tab (If plugin installed) - Run Scan 
Check violations for 'Accessible name must match or contain the visible label text.'

1. Step one
2. Step two
3. Step three
4. etc.

## Additional information

* Screenshots or code
![Screenshot 2019-09-02 at 18 07 15](https://user-images.githubusercontent.com/9660496/64128769-28906280-cdb0-11e9-8a4d-c5334dd4154d.png)

* Notes

**Comment by cal-smith** [comment]
Thanks for catching this! I'll try and roll this into #725 👍
